internal/controller/ipaddressclaim_controller.go

```diff
@@ -137,9 +137,17 @@ func (r *IpAddressClaimReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 					},
 				})
 			if err != nil {
-				return ctrl.Result{}, err
+				setConditionErr := r.SetConditionAndCreateEvent(ctx, o, netboxv1.ConditionIpAssignedFalse, corev1.EventTypeWarning, err.Error())
+				if setConditionErr != nil {
+					return ctrl.Result{}, fmt.Errorf("error updating status: %w, when getting an available IP address failed: %w", setConditionErr, err)
+				}
+
+				// since it's a Netbox error and not reconcile error, the log is of type Info and not Error
+				debugLogger.Info(fmt.Sprintf("error in getting available IP address: %s", err.Error()))
```

Comment:
The event will already create a log of the error. We could skip this log with the dubugLogger.